godot-core/src/builtin/script.rs

```diff
@@ -0,0 +1,875 @@
+    /// Name of the new class the script implements.
+    fn class_name(&self) -> GString;
+
+    /// Property setter for Godots virtual dispatch system. The engine will call this function when it wants to change a property on the script.
+    fn set(&mut self, name: StringName, value: &Variant) -> bool;
+
+    /// Property getter for Godots virtual dispatch system. The engine will call this function when it wants to read a property on the script.
+    fn get(&self, name: StringName) -> Option<Variant>;
+
+    /// A list of all the properties a script exposes to the engine.
+    fn property_list(&self) -> &[PropertyInfo];
+
+    /// A list of all the methods a script exposes to the engine.
+    fn method_list(&self) -> &[MethodInfo];
+
+    /// Method invoker for Godots virtual dispatch system. The engine will call this function when it wants to call a method on the script.
+    ///
+    /// All method calls are taking a mutable reference of the script instance, as the engine does not differentiate between immutable and
+    /// mutable method calls like rust.
+    ///
+    /// It's important that the script does not cause a second call to this function while executing a method call. This would result in a panic.
+    fn call(
+        &mut self,
+        method: StringName,
+        args: &[&Variant],
+    ) -> Result<Variant, sys::GDExtensionCallErrorType>;
+
+    /// Identifies the script instance as a placeholder. If this function and
+    /// [IScriptExtension::is_placeholder_fallback_enabled](crate::engine::IScriptExtension::is_placeholder_fallback_enabled) return true,
+    /// Godot will call [`Self::property_set_fallback`] instead of [`Self::set`].
+    fn is_placeholder(&self) -> bool;
+
+    /// Validation function for the engine to verify if the script exposes a certain method.
+    fn has_method(&self, method: StringName) -> bool;
+
+    /// Lets the engine get a reference to the script this instance was created for.
+    ///
+    /// This function has to return a reference, because Scripts are reference counted in Godot and it has to be guaranteed that the object is
+    /// not freed before the engine increased the reference count. (every time a `Gd<T>` which contains a reference counted object is dropped the
+    /// reference count is decremented.)
+    fn get_script(&self) -> &Gd<Script>;
+
+    /// Lets the engine fetch the type of a particular property.
+    fn get_property_type(&self, name: StringName) -> VariantType;
+
+    /// String representation of the script instance.
+    fn to_string(&self) -> GString;
+
+    /// Lets the engine get the owner / base object the script is attached to.
+    fn owner(&self) -> Gd<Object>;
+
+    /// A dump of all property names and values that are exposed to the engine.
+    fn property_state(&self) -> Vec<(StringName, Variant)>;
+
+    /// Lets the engine get a reference to the [`ScriptLanguage`] this instance belongs to.
+    fn language(&self) -> Gd<ScriptLanguage>;
```

Comment:
Maybe these 3 could have `get_` prefixes, for consistency with [`IScriptExtension::get_language`](https://godot-rust.github.io/docs/gdext/master/godot/engine/trait.IScriptExtension.html) and clarity (e.g. other methods starting with `property_*`)

Reply:
I removed `fn owner` as the FFI function was already removed before and I just forgot to remove it from the trait as well. The others are renamed.